lib/cn/cn_tree.c

```diff
@@ -2354,6 +2356,84 @@ cn_tree_fini(void)
     }
 }
 
+static int
+kblk_compare(const void *a, const void *b)
+{
+    const struct kvset_kblk *kblk_a = a, *kblk_b = b;
+
+    /* Return the kblock with the largest min key. */
+    return -1 * key_disc_cmp(&kblk_a->kb_kdisc_min, &kblk_b->kb_kdisc_min);
```

Comment:
Probably worth double checking this.

Reply:
i think key_disc_cmp() only does a fast-compare on the first 32-bytes, if keys are longer you need to do a full compare.  there's probably some code nearby a key_disc_cmp() call somewhere..  been a while, dunno...